src/hooks/useBlocksInfo.ts

```diff
@@ -26,41 +27,36 @@ interface Props {
 }
 
 const useBlocksInfo = ({ isApiReady, api, chain }: Props) => {
-  const [bestBlock, setBestBlock] = useState('');
-  const [bestBlockFinalized, setBestBlockFinalized] = useState('');
+  const [bestBlock, setBestBlock] = useMountedState('');
+  const [bestBlockFinalized, setBestBlockFinalized] = useMountedState('');
 
   useEffect(() => {
-    if (!api || !isApiReady || !chain) {
-      setBestBlock('');
-      setBestBlockFinalized('');
-      return;
-    }
-
-    let unsubscribeBestNumber: () => void;
-    api.derive.chain
-      .bestNumber((res) => {
-        setBestBlock(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumber = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
-
-    let unsubscribeBestNumberFinalized: () => void;
-    api.derive.chain
-      .bestNumberFinalized((res) => {
-        setBestBlockFinalized(res.toString());
-      })
-      .then((unsub) => {
-        unsubscribeBestNumberFinalized = unsub;
-      })
-      .catch((e) => logger.error('error reading blocks', e));
+    let unsubscribeBestNumber: Promise<VoidFn>;
+    let unsubscribeBestNumberFinalized: Promise<VoidFn>;
+    if (isApiReady && chain) {
+      try {
+        unsubscribeBestNumber = api.derive.chain.bestNumber((res) => {
+          setBestBlock(res.toString());
+        });
 
+        unsubscribeBestNumberFinalized = api.derive.chain.bestNumberFinalized((res) => {
+          setBestBlockFinalized(res.toString());
+        });
+      } catch (e) {
+        logger.error('error reading blocks', e);
+      }
+    }
     return function cleanup() {
-      unsubscribeBestNumberFinalized && unsubscribeBestNumberFinalized();
-      unsubscribeBestNumber && unsubscribeBestNumber();
+      unsubscribeBestNumber
+        .then((u) => {
+          u();
+        })
+        .catch((e) => logger.error('error unsubscribing bestNumber', e));
+      unsubscribeBestNumberFinalized
```

Comment:
This may still fail, because if there is an exception in the first call, the second promise might not be initialized. Worth adding `?` at the end and changing the type accordingly - IMHO best to just initialize the variables with `null` initially, so that the compiler can tell you that they might not be set.

Reply:
Also we run the code conditionally (`if` statement), so it can be `null` if the conditions are not met.

Reply:
fair point ... i will create a separate effect for each so each effect is in charge on one concern only.

Reply:
I'm not a fan of blanket-applying some rules. Of course it makes sense to separate concerns in different `useEffects`, but we should also avoid code repetition - if we do a lot of related stuff and we would need to pass the same dependencies to multiple `useEffects` or copy paste some code - it's simply boilerplate.

IMHO less amount of code is simply less bugs and more readability. This particular case is imho somewhere at the border, I would probably accept both - two separate hooks or a single one, but I feel refactoring this as part of that PR is just introducing noise - like I don't see anything wrong with the previous code.

Reply:
I think this change is necessary because it keeps consistency across other subscription hooks. But the try and catch issue must be solved, i think that adding another useEffect also helps to make the code easier to read.